Sources/LanguageServerProtocol/SupportTypes/Diagnostic.swift

```diff
@@ -111,11 +127,48 @@ extension DiagnosticCode: Codable {
   }
 }
 
-extension DiagnosticCode: CustomStringConvertible {
+extension DiagnosticCode: Codable {
+  enum ComplexCodingKeys: CodingKey {
+    case value
+    case target
+  }
+
+  public init(from decoder: Decoder) throws {
+    if let simpleCodeValue = try? NumberOrString(from: decoder) {
+      self.value = simpleCodeValue
+    } else {
+      let container = try decoder.container(keyedBy: ComplexCodingKeys.self)
+      self.value = try container.decode(NumberOrString.self, forKey: .value)
+      self.target = try container.decode(URL.self, forKey: .target)
```

Comment:
I tend to agree. For now I changed this to ensure we can decode `{ value : "foo" }` without a target. Since the spec isn't shipped yet I might bring this up as a possible change on the LSP side of things.